lib/client/Client.js

```diff
@@ -192,8 +214,10 @@ export default class Client {
    */
   authPassword(username, password, deviceName) {
     return this.authenticate(username, deviceName, function() {
-      return { method: 'password', password: password };
-    });
+      return this.transport.promiseProvider.newPromise(function(resolve, reject) {
```

Comment:
Excellent change, that makes `authPassword` more consistent with `authExternal` in the way they use promises.

As a side note, I won't mind if you want to keep a reference to the promiseProvider in the client instance directly so that you can use `this.promiseProvider` directly.